beater/api/mux.go

```diff
@@ -196,20 +187,30 @@ func apmMiddleware(m map[request.ResultID]*monitoring.Int) []middleware.Middlewa
 	}
 }
 
-func backendMiddleware(_ *config.Config, auth *authorization.Handler, m map[request.ResultID]*monitoring.Int) []middleware.Middleware {
-	return append(apmMiddleware(m),
-		middleware.AuthorizationMiddleware(auth, true))
+func backendMiddleware(cfg *config.Config, auth *authorization.Handler, m map[request.ResultID]*monitoring.Int) []middleware.Middleware {
+	backendMiddleware := append(apmMiddleware(m),
+		middleware.AuthorizationMiddleware(auth, true),
+	)
+	if cfg.AugmentEnabled {
+		backendMiddleware = append(backendMiddleware, middleware.SystemMetadataMiddleware())
```

Comment:
Having `middleware.SystemMetadataMiddleware()` take a boolean argument would push complexity down and keep this more streamlined and also in line of how other middlewares work. Something like: 
```golang
backendMiddleware := append(apmMiddleware(m),
		middleware.AuthorizationMiddleware(auth, true),
                middleware.SystemMetadataMiddleware(cfg.AugmentEnabled)
	)
```

```golang
func SystemMetadataMiddleware(enabled bool) Middleware {
	return func(h request.Handler) (request.Handler, error) {
		if !enabled {
			return h, nil
		}
                // stuff
```

Same for others.

Reply:
To me it seems strange to install middleware when you know it will _never_ have any use. This configuration is static, not based on any property of the request.

If you know that the middleware isn't even installed, you can immediately put it out of your mind what it does. If it's installed but takes in config, then you still need to keep in mind its behviour.

> also in line of how other middlewares work

I can see two middlewares which take boolean params, but they don't mean enable/disable.
- AuthorizationMiddleware _always_ creates an Authorization object for the request, but optionally enforces the authorization up front
- KillSwitchMiddleware either runs the request or returns an error if the _route_ is disabled

Reply:
> If it's installed but takes in config, then you still need to keep in mind its behviour

fair enough :+1: